### PR TITLE
limit process instances access to initiator

### PR DESCRIPTION
**Pull request by balsarori**
Ref: https://github.com/Activiti/Activiti/issues/3838

### CODE REVIEW

File: activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/security/ProcessInstanceRestrictionService.java [hunk 1]
```diff
@@ -27,24 +30,40 @@
...
-        return restrictionBuilder.applyProcessDefinitionKeyFilter(predicate,
+        return restrictionBuilder.applyProcessDefinitionKeyFilter(initiatorPredicate,
                                                  securityPolicyAccess,
                                                  processInstanceFilter);
     }
 
+    private Predicate applyInitiatorRestriction(Predicate predicate) {
+        String userId = securityManager.getAuthenticatedUserId();
+
+        if (userId == null) {
```
Comment (balsarori) [inline activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/security/ProcessInstanceRestrictionService.java:61]: I'm not sure if there could be a scenario where the current user is null. I followed the same check here --> https://github.com/Activiti/activiti-cloud/blob/develop/activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/security/TaskLookupRestrictionService.java#L102

**Review by Giovanni007 — APPROVED** [review]
LGTM!

File: activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/ProcessInstanceController.java [hunk 2]
```diff
@@ -149,4 +147,10 @@ public ProcessInstanceController(ProcessInstanceRepository processInstanceReposi
                                                                                     pageable),
                                                   processInstanceRepresentationModelAssembler);
     }
+
+    private boolean canRead(ProcessInstanceEntity processInstanceEntity) {
```
Comment (mauriziovitale) [inline activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/ProcessInstanceController.java:151]: what I don't like is that the canRead is not giving any clue about the initiator

**Review by mauriziovitale — APPROVED** [review]

File: activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/ProcessInstanceController.java (hunk 2 above)
Comment (balsarori) [reply activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/ProcessInstanceController.java:151]: user needs to be the process initiator and have permissions to read the app `securityPoliciesApplicationService.canRead`

**Pull request merged**